### PR TITLE
⚠️ Do not merge into main

### PR DESCRIPTION
**Pull request by jeremyf**
This job really should be in Bulkrax, but I digress.  I've switched this to a job (from a service) to reflect that we want to submit this potentially long-running job.

Related to:

- https://github.com/scientist-softserv/adventist-dl/issues/430
- https://github.com/scientist-softserv/adventist-dl/issues/437-

### CODE REVIEW

**Comment by jeremyf** [comment]
We want to launch the following:

```
Account.switch!('sdapi') 
RerunErroredEntriesForImporterJob.perform_later(importer_id: 34)
```

**Comment by orangewolf** [comment]
for the record, this got run

**Comment by ShanaLMoore** [comment]
```
switch!'sdapi'
RerunErroredEntriesForImporterJob.new.perform(importer_id: 34)
```

File: app/jobs/rerun_errored_entries_for_importer_job.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  def do_it!
+    reimport_logging_context = "#{importer.class} ID=#{importer.id} with #{last_run.class} ID=#{last_run.id}"
+    relation = last_run.statuses.where(statusable_type: 'Bulkrax::Entry')
+
+    if error_classes.empty?
+      logger.info("Starting re-importing #{reimport_logging_context} with entries that had any error.")
+      relation = relation.where.not(error_class: nil)
+    else
+      # rubocop:disable Metrics/LineLength
+      logger.info("Starting re-importing #{reimport_logging_context} with entries that had the following errors: #{error_class.inspect}.")
```
Comment (ShanaLMoore) [inline app/jobs/rerun_errored_entries_for_importer_job.rb:44]:
```suggestion
      logger.info("Starting re-importing #{reimport_logging_context} with entries that had the following errors: #{error_classes.inspect}.")
```

File: app/jobs/rerun_errored_entries_for_importer_job.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @error_classes = Array.wrap(error_classes).map(&:to_s)
+
+    do_it!
+  end
+
+  attr_reader :importer, :last_run, :new_run, :logger, :error_classes
+
+  def do_it!
+    reimport_logging_context = "#{importer.class} ID=#{importer.id} with #{last_run.class} ID=#{last_run.id}"
+    relation = last_run.statuses.where(statusable_type: 'Bulkrax::Entry')
```
Comment (ShanaLMoore) [inline app/jobs/rerun_errored_entries_for_importer_job.rb:37]: We have a problem. when we re run an entry, it does not update the last_run's failed count. so last_run is not an accurate way to collect all of the current errors. 🤔

**Review by ShanaLMoore — APPROVED** [review]

**Comment by ShanaLMoore** [comment]
ref this PR instead: https://github.com/scientist-softserv/adventist-dl/pull/448/files

**Pull request closed without merging**